### ISSUE TITLE
For test_datagen.flow from directory, what should I specify the class_mode?

### ISSUE THREAD

**Issue by RezuanaHaque** [keras-team/keras#16913]
I'm attempting to create a CNN model using image data. My entire data set was divided into train, test, and validation. I used class_mode = 'categorical' for the training data since training data needs labels, and class_mode = None for the test data because testing shouldn't be done with labels. the snippet of code
```
train_datagen =  ImageDataGenerator(
        rescale=1 / 255.0,
        rotation_range=20,
        zoom_range=0.05,
        width_shift_range=0.05,
        height_shift_range=0.05,
        shear_range=0.05,
        horizontal_flip=True,
        vertical_flip=True,
        fill_mode="nearest"
    )

test_datagen = ImageDataGenerator(
      rescale=1/255,
)

train_generator = train_datagen.flow_from_directory(
    directory=train_dir,
    target_size=(height, width),
    batch_size=batch_size,
    class_mode='categorical',
    shuffle=True,
    seed=42
)

valid_generator = train_datagen.flow_from_directory(
    directory=val_dir,
    target_size=(height, width),
    batch_size=batch_size,
    class_mode='categorical',
    shuffle=True,
    seed=42
)


test_generator = test_datagen.flow_from_directory(
    "/content/drive/MyDrive/MonkeyPox/output/test",
    target_size=(height, width),
    batch_size=1,
    class_mode=None,   **here**
    shuffle=False,
    seed=42
)
```

when I specify `class_moede=None` in the `test_generator`,

`model.evaluate(test_generator, batch_size=16)`
the loss & accuracy is [0.0, 0.0], which means my model is failed to evaluate

again when I specify `class_moede='categorical'` in the `test_generator`,

`model.evaluate(test_generator, batch_size=16)`
it gives a certain loss and accuracy

My question is why i am getting [0.0, 0.0] when i am using `class_mode=None` in the `test_generator` ?

**Comment by tilakrayal** [comment]
@RezuanaHaque,
To expedite the trouble-shooting process, could you please provide a complete code and the TensorFlow version you are using. Thank you!
